tests/core/pyspec/eth2spec/test/gloas/block_processing/test_process_attestation.py

```diff
@@ -0,0 +1,489 @@
+    """
+    # Advance to slot 5 (only creates one block at slot 5)
+    transition_to_slot_via_block(spec, state, 5)
+
+    # Create a custom arbitrary block root that's guaranteed to be different
+    custom_block_root = spec.Root(b"\x01" * 32)  # Different from any real block root
+    attestation = get_valid_attestation(spec, state, slot=4, beacon_block_root=custom_block_root)
+
+    attestation.data.index = 1
+
+    sign_attestation(spec, state, attestation)
+
+    # Verify the intended blockroot conditions
+    is_matching_blockroot = attestation.data.beacon_block_root == spec.get_block_root_at_slot(
+        state, spec.Slot(attestation.data.slot)
+    )
+    is_current_blockroot = attestation.data.beacon_block_root != spec.get_block_root_at_slot(
+        state, spec.Slot(attestation.data.slot - 1)
+    )
+    assert is_matching_blockroot is False, "Expected is_matching_blockroot = False"
+    assert is_current_blockroot is True, "Expected is_current_blockroot = True"
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is False
+    yield from run_attestation_processing(spec, state, attestation, valid=True)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_invalid_same_slot_attestation_index_one(spec, state):
+    """
+    Test that same-slot condition with index = 1 is invalid (same-slot must use index = 0)
+    """
+    transition_to_slot_via_block(spec, state, 1)
+    transition_to_slot_via_block(spec, state, 2)
+    transition_to_slot_via_block(spec, state, 3)
+    next_slots(spec, state, spec.MIN_ATTESTATION_INCLUSION_DELAY)
+
+    attestation_slot = 2
+    attestation = get_valid_attestation(spec, state, slot=attestation_slot)
+
+    attestation.data.beacon_block_root = spec.get_block_root_at_slot(state, attestation_slot)
+
+    attestation.data.index = 1
+
+    sign_attestation(spec, state, attestation)
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is True
+    yield from run_attestation_processing(spec, state, attestation, valid=False)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_valid_attestation_data_index_zero_previous_slot(spec, state):
+    """
+    Test that attestation with index = 0 is valid in Gloas for previous slot attestations
+    """
+    transition_to_slot_via_block(
+        spec, state, 5
+    )  # Advance to slot 5 (only creates one block at slot 5)
+
+    slot_3_block_root = spec.get_block_root_at_slot(state, 3)
+    attestation = get_valid_attestation(spec, state, slot=4, beacon_block_root=slot_3_block_root)
+
+    attestation.data.index = 0
+
+    sign_attestation(spec, state, attestation)
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is False
+    yield from run_attestation_processing(spec, state, attestation, valid=True)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_valid_attestation_data_index_zero_same_slot(spec, state):
```

Comment:
Same for this, the same setup as for index 1 would work, and then you just test with a separate attestation. 

If renaming the tests accordingly is an issue, then just copying the test is good.

Reply:
I think it's already copy/pasted to `test_invalid_same_slot_attestation_index_one`, the only difference is index and assert to false